### PR TITLE
migrate from dep to Go modules

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/10177
Needed for https://github.com/giantswarm/ingress-exporter/pull/10 (Circle builds won't pass otherwise)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	github.com/giantswarm/microerror v0.2.0
+	github.com/giantswarm/microkit v0.2.1
+	github.com/giantswarm/micrologger v0.3.1
+	github.com/giantswarm/operatorkit v0.2.0
+	github.com/prometheus/client_golang v1.3.0
+	github.com/spf13/viper v1.6.3
+	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
+	k8s.io/api v0.16.6
+	k8s.io/apimachinery v0.16.6
+	k8s.io/client-go v0.16.6
```
Comment (kopiczko) [inline go.mod:19]: [The latest aws-operator uses 0.17.2](https://github.com/giantswarm/aws-operator/blob/5da9ed7374390cdeb031dfeec85e03da8c7dfc40/go.mod#L41) so I guess we should use that here too .

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191114103820-f023614fb9ea
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191114111510-6d1ed697a64b
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191114110717-50a77e50d7d9
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191114111229-2e90afcb56c7
+	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191114113550-6123e1c827f7
+	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191114110954-d67a8e7e2200
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191114112655-db9be3e678bb
+	k8s.io/metrics => k8s.io/metrics v0.0.0-20191114105837-a4a2842dc51b
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191114104439-68caf20693ac
+)
```
Comment (kopiczko) [inline go.mod:44]: Do we need all that garbage?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/ingress-exporter
+!/helm/ingress-exporter
```
Comment (kopiczko) [inline .gitignore:2]: I think this line is redundant.

```suggestion 
```

File: go.mod (hunk 2 above)
Comment (wtty-fool) [reply go.mod:44]: I can trim it down to the 2 repos we actually use :+1:

File: go.mod (hunk 1 above)
Comment (wtty-fool) [reply go.mod:19]: Sure

Comment (tfussell) [reply go.mod:19]: There's a `replace` directive in the `go.mod` below which looks like it's pointing to 1.16.3. We should probably fix this so everything is using 1.16.6 for consistency.

Comment (tfussell) [reply go.mod:19]: See https://github.com/kubernetes/client-go/commit/6c5935290e33 (from pseudo version)

Comment (wtty-fool) [reply go.mod:19]: I'll make sure it's consistent and using `1.16.6` then.

Comment (wtty-fool) [reply go.mod:19]: Oh, I actually have it replaced to the same version as aws-op, nevermind then.

**Pull request merged**